supplementary/alica_ros_turtlesim/src/base.cpp

```diff
@@ -24,8 +24,10 @@ Base::Base(ros::NodeHandle& nh, ros::NodeHandle& priv_nh, const std::string& nam
     ALICATurtleWorldModel::init(nh, priv_nh);
     // Initialize Alica
     ac = new alica::AlicaContext(AlicaContextParams(name, path + "/etc/", roleset, master_plan, false));
-    ac->setCommunicator<alicaRosProxy::AlicaRosCommunication>();
-    ac->setTimerFactory<alicaRosTimer::AlicaRosTimerFactory>(4);
+
+    ros::AsyncSpinner spinner(0);
+    ac->setCommunicator<alicaRosProxy::AlicaRosCommunication>(spinner);
```

Comment:
The callback queue & async spinner's lifetime should be greater than the lifetime of the engine which won't be the case here. Make the spinner a member variable of base class & ensure it is constructed before the context & destroyed after the context is destroyed

Reply:
Got it.
That also means i should not stop the spinner in the timer/communication's desctructor, right?

Reply:
yes